components/builder-protocol/src/originsrv.rs

```diff
@@ -744,10 +744,14 @@ impl From<hab_core::package::PackageIdent> for OriginPackageIdent {
         ident.set_origin(value.origin);
         ident.set_name(value.name);
         if let Some(ver) = value.version {
-            ident.set_version(ver);
+            if ver != "" {
```

Comment:
Is this check here primarily to work around for bad data in the db?

Reply:
exactly, since we could have potentially poisoned the db with the other bug fixed in here

Reply:
would be good to also clean up the db after